R/rownames.R

```diff
@@ -65,6 +67,23 @@ rownames_to_column <- function(df, var = "rowname") {
   new_df
 }
 
+#' @export
+#' @rdname rownames
+rowid_to_column <- function(df, var = "rowid") {
+  stopifnot(is.data.frame(df))
+  
+  if (has_name(df, var))
+    stopc("There is a column named ", var, " already!")
+  
+  if(nrow(df) == 0)
+    stopc('Data frame must have 1 or more rows')
+  
+  rn <- tibble(seq(1, nrow(df)))
+  names(rn) <- var
+  new_df <- cbind(rn, df)
```

Comment:
This code, and its equivalent in `rownames_to_column()`, really should be using `add_column()` via `add_column(data, !!(var) := rn, ...)` or perhaps with `invoke()`, see also `?rlang::invoke` and  `?"!!"`:

``` r
x <- "y"; tibble::tibble(!!(x) := 5)
#> # A tibble: 1 × 1
#>       y
#>   <dbl>
#> 1     5
```